### ISSUE TITLE
How to run algorithms where there isn't a need for dataloaders?

### ISSUE THREAD

**Issue by nsidn98** [PyTorchLightning/pytorch-lightning#2182]
#### What is your question?
In on-policy algorithms in reinforcement learning, rollouts are generated on the fly and there is no need for a replay buffer and consequently a dataloader. In these cases, the loss function is calculated according to the current states obtained (generally by using multiple parallel environments). 

Hence, is it possible to bypass the dataloader requirement for lightning?

**Comment by williamFalcon** [comment]
Check these examples out:

https://github.com/djbyrne/core_rl

**Comment by nsidn98** [comment]
@williamFalcon the examples in the linked repository does not have on-policy algorithms. Is there a way to bypass the need of train_dataloader() in the lightning module?
